packages/contracts/src/resources/site.ts

```diff
@@ -176,19 +177,48 @@ export type MaybePubOptions = {
 	withValues?: boolean;
 };
 
+/**
+ * Information about the field that the value belongs to.
+ */
+type ValueFieldInfo = {
+	schemaName: CoreSchemaType;
+	fieldId: PubFieldsId;
+	fieldSlug: string;
+	fieldName: string;
+};
+
 type ValueBase = {
 	id: PubValuesId;
-	fieldId: PubFieldsId;
 	value: unknown;
 	createdAt: Date;
 	updatedAt: Date;
-	/**
-	 * Information about the field that the value belongs to.
-	 */
-	schemaName: CoreSchemaType;
-	fieldSlug: string;
-	fieldName: string;
-};
+} & ValueFieldInfo;
+
+type ValuesWithFormElements =
+	| // With both values and form elements
+	(ValueBase & {
+			formElementId: FormElementsId;
+			formElementLabel: string | null;
+			formElementConfig:
+				| { label?: string }
+				| { relationshipConfig: { label?: string } }
+				| null;
```

Comment:
maybe there's a better way to get this config type?

Reply:
feels like there should be but idk. thanks for writing this out!